src/mango/src/mango_cursor_view.erl

```diff
@@ -441,7 +496,12 @@ doc_member(Cursor, RowProps) ->
             case mango_util:defer(fabric, open_doc, [Db, Id, Opts]) of
                 {ok, #doc{} = DocProps} ->
                     Doc = couch_doc:to_json_obj(DocProps, []),
-                    match_doc(Selector, Doc, ExecutionStats1);
+                    case match_and_extract_doc(Doc, Selector, Cursor#cursor.fields) of
+                        {match, FinalDoc} ->
+                            {ok, FinalDoc, {execution_stats, ExecutionStats1}};
+                        {no_match, _} ->
```

Comment:
Should we assert `{no_match, undefined}` since we know it should be only that as the `no_match` choice?

Reply:
There's no reason to leave it unspecified. I wasn't sure whether to go for more "fixed" pattern matching or be freer, so I'm more than happy to change. I'll update and squash in a new commit.